perl-lib/OESS/lib/OESS/Cloud.pm

```diff
@@ -0,0 +1,119 @@
+package OESS::Cloud;
+
+use strict;
+use warnings;
+
+use Exporter;
+
+use OESS::Cloud::AWS;
+
+
+=head2 setup_endpoints
+
+setup_endpoints configures cloud services for any interface in
+C<$endpoints> with a configured cloud interconnect id. Once complete,
+any information related to new cloud services is recorded in the
+resulting endpoint list. The resulting endpoint list should be used as
+a replacement for the parent VRF's endpoints.
+
+    my $setup_endpoints = Cloud::setup_endpoints('vrf1', $vrf->endpoints, 123456789);
+    $vrf->endpoints($setup_endpoints);
+    $vrf->update_db();
+
+=cut
+sub setup_endpoints {
+    my $vrf_name   = shift;
+    my $endpoints  = shift;
+    my $result     = [];
+
+    foreach my $ep (@$endpoints) {
+        if (!$ep->interface()->cloud_interconnect_id) {
+            push @$result, $ep;
+            next;
+        }
+
+        warn "oooooooooooo AWS oooooooooooo";
+        my $aws = OESS::Cloud::AWS->new();
+
+        if ($ep->interface()->cloud_interconnect_type eq 'aws-hosted-connection') {
+            my $res = $aws->allocate_connection(
```

Comment:
for both types... can't AWS give us the results from their side, enough so that people don't need to specify tag, bandwidth, etc... and I guess I don't see where we get the IP / ASN back from it... we should be able to get that right?